src/ngraph/serializer.cpp

```diff
@@ -1608,7 +1610,15 @@ shared_ptr<Node> JSONDeserializer::deserialize_node(json node_js)
         case OP_TYPEID::Product:
         {
             auto reduction_axes = deserialize_axis_set(node_js.at("reduction_axes"));
-            node = make_shared<op::Product>(args[0], reduction_axes);
+            if (op_version == 0)
+            {
+                node = make_shared<op::Product>(args[0], reduction_axes);
```

Comment:
`v0` actually supports dynamic `reduction_axes` (meaning the old logic was broken :grimacing:). For v0, can we check if `reduction_axes` is missing, and call the two-input constructor on `args[0], args[1]` in that case? On the serializer side we will always want to treat `reduction_axes` as an input rather than an attribute.

Reply:
OTOH, `v1` should _always_ treat `reduction_axes` as an input, so please make that change too.

Reply:
Done
- pass `args[1]` when `reduction_axes` is empty (not serialized) for `v0::Product`
- pass `args[1]` always for `v1::ReduceProd`